karapace/karapace.py

```diff
@@ -7,28 +7,19 @@
 
 from functools import partial
 from http import HTTPStatus
+from karapace.config import Config
 from karapace.rapu import HTTPResponse, RestApp
 from typing import NoReturn, Union
 
 import asyncio
 import logging
-import os
 
 
 class KarapaceBase(RestApp):
-    def __init__(self, config: dict) -> None:
-        sentry_config = config.get("sentry", {"dsn": None}).copy()
-        super().__init__(app_name="karapace", sentry_config=sentry_config)
+    def __init__(self, config: Config) -> None:
+        super().__init__(app_name="karapace", config=config)
 
         self.kafka_timeout = 10
-        self.config = config
-        self._sentry_config = sentry_config
-        if os.environ.get("SENTRY_DSN"):
-            self._sentry_config["dsn"] = os.environ["SENTRY_DSN"]
-        if "tags" not in self._sentry_config:
-            self._sentry_config["tags"] = {}
-        self._sentry_config["tags"]["app"] = "Karapace"
```

Comment:
This is what was hard to understand, `self._sentry_config` is never used, only modified. Turns out that `self._sentry_config` is the same object as `config["sentry"]` and that makes its way to the stats client, so that object over there was relying on this spooky action at a distance